src/filters/capture/regex.rs

```diff
@@ -0,0 +1,33 @@
+use crate::metadata::Value;
+
+use super::Metrics;
+
+/// Capture from the start of the packet.
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+pub struct Regex {
+    /// The regular expression to use for capture.
+    #[serde(with = "serde_regex")]
+    pub regex: regex::bytes::Regex,
+}
+
+impl super::CaptureStrategy for Regex {
+    fn capture(&self, contents: &mut Vec<u8>, _metrics: &Metrics) -> Option<Value> {
+        let matches = self
+            .regex
+            .find_iter(contents)
+            .map(|mat| Value::Bytes(bytes::Bytes::copy_from_slice(mat.as_bytes())))
+            .collect::<Vec<_>>();
+
+        if matches.len() > 1 {
+            Some(Value::List(matches))
+        } else {
+            matches.into_iter().next()
+        }
```

Comment:
this looks like regex can potentially capture a list or single value? if so we should document the behavior (or maybe just return a singleton list in the latter so its consistent?)